src/native-stack/views/NativeStackView.tsx

```diff
@@ -268,6 +268,10 @@ const RouteView = ({
   const Screen = React.useContext(ScreenContext);
   const { dark } = useTheme();
 
+  if (statusBarStyle === 'auto') {
```

Comment:
Maybe some warning that the behavior changed and you should use `system`? Or am I missing something?

Reply:
I think adding some note in the changelog should be sufficient - especially, if we want to merge this for v4 😄